File: Test12/step_definitions/Seamless Purchase Completion.js
import { Given, When, Then } from "cypress-cucumber-preprocessor/steps";
import PurchasePage from "../pages/purchasePage";
import PaymentPage from "../pages/paymentPage";
import ConfirmationPage from "../pages/confirmationPage";
import ErrorHandlingPage from "../pages/errorHandlingPage";
import SessionSyncPage from "../pages/sessionSyncPage";
import HistoryPage from "../pages/historyPage";
import SecurityPage from "../pages/securityPage";
import IntegrationPage from "../pages/integrationPage";
import FeedbackPage from "../pages/feedbackPage";

Given("a user is on the purchase page on desktop", () => {
 PurchasePage.visitDesktop();
});

When("the user completes the purchase process", () => {
 PurchasePage.completePurchase();
});

Then("the purchase flow should be identical on mobile", () => {
 PurchasePage.verifyMobileFlow();
});

Given("a user is on the payment page on desktop", () => {
 PaymentPage.visitDesktop();
});

When("the user selects a payment method and completes the transaction", () => {
 PaymentPage.selectPaymentMethod();
 PaymentPage.completeTransaction();
});

Then("the payment gateway should function consistently across platforms", () => {
 PaymentPage.verifyGatewayConsistency();
});

Given("a user completes a purchase on desktop", () => {
 ConfirmationPage.completePurchaseDesktop();
});

When("the transaction is successful", () => {
 ConfirmationPage.verifyTransactionSuccess();
});

Then("the user should receive instant confirmation on their purchase", () => {
 ConfirmationPage.verifyInstantConfirmation();
});

Given("a user is on the payment page", () => {
 PaymentPage.visit();
});

When("the user chooses a payment method", () => {
 PaymentPage.choosePaymentMethod();
});

Then("multiple payment methods should be supported", () => {
 PaymentPage.verifyMultipleMethodsSupport();
});

Given("a user encounters an error during purchase on desktop", () => {
 ErrorHandlingPage.encounterErrorDesktop();
});

When("the same error occurs on mobile", () => {
 ErrorHandlingPage.encounterErrorMobile();
});

Then("error handling should be consistent across devices", () => {
 ErrorHandlingPage.verifyErrorConsistency();
});

Given("a user is logged in on desktop", () => {
 SessionSyncPage.loginDesktop();
});

When("the user switches to mobile", () => {
 SessionSyncPage.switchToMobile();
});

Then("the user session should sync between desktop and mobile", () => {
 SessionSyncPage.verifySessionSync();
});

Given("a user completes a purchase", () => {
 HistoryPage.completePurchase();
});

When("the user checks their purchase history on any device", () => {
 HistoryPage.checkPurchaseHistory();
});

Then("the purchase history should be accessible on all devices", () => {
 HistoryPage.verifyHistoryAccessibility();
});

Given("a user is making a purchase", () => {
 SecurityPage.initiatePurchase();
});

When("security protocols are applied", () => {
 SecurityPage.applySecurityProtocols();
});

Then("security protocols must be consistent across platforms", () => {
 SecurityPage.verifySecurityConsistency();
});

Given("a user uses a third-party service during purchase", () => {
 IntegrationPage.useThirdPartyService();
});

When("the integration is initiated", () => {
 IntegrationPage.initiateIntegration();
});

Then("the integration with third-party services must be seamless", () => {
 IntegrationPage.verifyIntegrationSeamlessness();
});

Given("user feedback is collected", () => {
 FeedbackPage.collectFeedback();
});

When("analyzing feedback for improvements", () => {
 FeedbackPage.analyzeFeedback();
});

Then("user feedback should guide improvements in purchase flow", () => {
 FeedbackPage.verifyGuidedImprovements();
});

File: Test12/step_definitions/Seamless Purchase Completion.java
package com.cucumber.steps;

import com.framework.cucumber.TestHarness;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import com.page_objects.PurchasePage;

public class PurchaseSteps extends PurchasePage {

    private TestHarness testHarness = new TestHarness();

    @Given("a user is on the purchase page on desktop")
    public void aUserIsOnThePurchasePageOnDesktop() {
        launchPurchasePage();
    }

    @When("the user completes the purchase process")
    public void theUserCompletesThePurchaseProcess() {
        completePurchaseProcess();
    }

    @Then("the purchase flow should be identical on mobile")
    public void thePurchaseFlowShouldBeIdenticalOnMobile() {
        verifyPurchaseFlowOnMobile();
    }

    @Given("a user is on the payment page on desktop")
    public void aUserIsOnThePaymentPageOnDesktop() {
        launchPaymentPage();
    }

    @When("the user selects a payment method and completes the transaction")
    public void theUserSelectsAPaymentMethodAndCompletesTheTransaction() {
        selectPaymentMethodAndCompleteTransaction();
    }

    @Then("the payment gateway should function consistently across platforms")
    public void thePaymentGatewayShouldFunctionConsistentlyAcrossPlatforms() {
        verifyPaymentGatewayConsistency();
    }

    @Given("a user completes a purchase on desktop")
    public void aUserCompletesAPurchaseOnDesktop() {
        completePurchaseOnDesktop();
    }

    @When("the transaction is successful")
    public void theTransactionIsSuccessful() {
        verifyTransactionSuccess();
    }

    @Then("the user should receive instant confirmation on their purchase")
    public void theUserShouldReceiveInstantConfirmationOnTheirPurchase() {
        verifyInstantPurchaseConfirmation();
    }

    @Given("a user is on the payment page")
    public void aUserIsOnThePaymentPage() {
        launchPaymentPage();
    }

    @When("the user chooses a payment method")
    public void theUserChoosesAPaymentMethod() {
        choosePaymentMethod();
    }

    @Then("multiple payment methods should be supported")
    public void multiplePaymentMethodsShouldBeSupported() {
        verifyMultiplePaymentMethodsSupport();
    }

    @Given("a user encounters an error during purchase on desktop")
    public void aUserEncountersAnErrorDuringPurchaseOnDesktop() {
        simulateErrorOnDesktop();
    }

    @When("the same error occurs on mobile")
    public void theSameErrorOccursOnMobile() {
        simulateErrorOnMobile();
    }

    @Then("error handling should be consistent across devices")
    public void errorHandlingShouldBeConsistentAcrossDevices() {
        verifyConsistentErrorHandling();
    }

    @Given("a user is logged in on desktop")
    public void aUserIsLoggedInOnDesktop() {
        loginUserOnDesktop();
    }

    @When("the user switches to mobile")
    public void theUserSwitchesToMobile() {
        switchToMobile();
    }

    @Then("the user session should sync between desktop and mobile")
    public void theUserSessionShouldSyncBetweenDesktopAndMobile() {
        verifySessionSync();
    }

    @Given("a user completes a purchase")
    public void aUserCompletesAPurchase() {
        completePurchase();
    }

    @When("the user checks their purchase history on any device")
    public void theUserChecksTheirPurchaseHistoryOnAnyDevice() {
        checkPurchaseHistory();
    }

    @Then("the purchase history should be accessible on all devices")
    public void thePurchaseHistoryShouldBeAccessibleOnAllDevices() {
        verifyPurchaseHistoryAccessibility();
    }

    @Given("a user is making a purchase")
    public void aUserIsMakingAPurchase() {
        initiatePurchase();
    }

    @When("security protocols are applied")
    public void securityProtocolsAreApplied() {
        applySecurityProtocols();
    }

    @Then("security protocols must be consistent across platforms")
    public void securityProtocolsMustBeConsistentAcrossPlatforms() {
        verifySecurityProtocolsConsistency();
    }

    @Given("a user uses a third-party service during purchase")
    public void aUserUsesAThirdPartyServiceDuringPurchase() {
        useThirdPartyService();
    }

    @When("the integration is initiated")
    public void theIntegrationIsInitiated() {
        initiateIntegration();
    }

    @Then("the integration with third-party services must be seamless")
    public void theIntegrationWithThirdPartyServicesMustBeSeamless() {
        verifySeamlessIntegration();
    }

    @Given("user feedback is collected")
    public void userFeedbackIsCollected() {
        collectUserFeedback();
    }

    @When("analyzing feedback for improvements")
    public void analyzingFeedbackForImprovements() {
        analyzeFeedback();
    }

    @Then("user feedback should guide improvements in purchase flow")
    public void userFeedbackShouldGuideImprovementsInPurchaseFlow() {
        guideImprovementsWithFeedback();
    }
}